File: DEMOAPPS/DEMOAPP001_TYPESCRIPT_CYPRESS/tests/screenplay/actors/SudokuActors.ts
import { Actor, Cast } from '@serenity-js/core';
import { UseSudokuSolver } from '../abilities/UseSudokuSolver';
import { LoadPuzzles } from '../abilities/LoadPuzzles';
import * as path from 'path';

const PUZZLES_PATH = path.resolve(__dirname, '../../../puzzles.json');

/**
 * SudokuActors: Cast implementation for the DEMOAPP001 Screenplay layer.
 *
 * Responsible for equipping each Actor with the right Abilities.
 * Single actor persona: "Solver" — recreated fresh per scenario via
 * Serenity/JS automatic crew reset.
 *
 * Design note: all Ability configuration lives here. Adding a new Ability
 * (e.g., WriteAuditLog) only requires changing this file.
 */
export const SudokuActors: Cast = {
  prepare(actor: Actor): Actor {
    return actor.whoCan(
      new UseSudokuSolver(),
      LoadPuzzles.from(PUZZLES_PATH),
    );
  },
};

File: demo-apps/demoapp003-csharp-specflow/tests/screenplay/actors/SudokuActors.cs
using DemoApp003.Specs.Screenplay.Abilities;
using DemoApp003.Specs.Screenplay.Support;

namespace DemoApp003.Specs.Screenplay.Actors;

public static class SudokuActors
{
    public const string SolverActor = "Solver";

    public static Actor MakeSolverActor() =>
        Actor.Named(SolverActor).WhoCan(new UseSudokuSolver(), LoadPuzzles.From("puzzles.json"));
}
